vars/onTerraformChangeInPR.groovy

```diff
@@ -1,15 +1,19 @@
 def call(Closure block) {
+  def credentialsId = env.GIT_CREDENTIALS_ID
   folderExists('infrastructure') {
     writeFile file: 'check-infrastructure-files-changed.sh', text: libraryResource('uk/gov/hmcts/infrastructure/check-infrastructure-files-changed.sh')
 
-    def infraFolderHasChanges = sh(
-      script: "chmod +x check-infrastructure-files-changed.sh\n" +
-        "    ./check-infrastructure-files-changed.sh",
-      returnStatus: true
-    )
-    sh 'rm check-infrastructure-files-changed.sh'
-    if (infraFolderHasChanges == 1) {
-      return block.call()
+    withCredentials([usernamePassword(credentialsId: credentialsId, passwordVariable: 'BEARER_TOKEN', usernameVariable: 'APP_ID')]) {
```

Comment:
I'm not sure where they're set as env vars but I assume it would just fallback to the current behaviour if somehow they're null.
I also haven't looked into withCredentials in detail to see how it works under the hood, just plugged it in here

Reply:
I can see `withCredentials` also called in checkTerraformFormat.groovy. there it also assumes credentialsId and BEAER_TOKEN are always set by Jenkins.